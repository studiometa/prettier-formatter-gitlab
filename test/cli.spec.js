import { describe, it, expect, afterEach } from 'bun:test';
import { existsSync, unlinkSync, readFileSync } from 'node:fs';
import { execSync } from 'node:child_process';

const CODE_QUALITY_FILENAME = 'gl-prettier-codequality.json';

afterEach(() => {
  if (existsSync(CODE_QUALITY_FILENAME)) {
    unlinkSync(CODE_QUALITY_FILENAME);
  }
});

describe('prettier-formatter-gitlab cli', () => {
  it('should create a code quality report file', () => {
    try {
      execSync(
        `PRETTIER_CODE_QUALITY_REPORT="${CODE_QUALITY_FILENAME}" ./bin/cli.js "prettier -l test"`,
      );
    } catch (err) {
      // Silence is golden.
    }

    expect(existsSync(CODE_QUALITY_FILENAME)).toBe(true);
    const content = readFileSync(CODE_QUALITY_FILENAME);
    expect(content.toString()).toMatchSnapshot();
  });
});
