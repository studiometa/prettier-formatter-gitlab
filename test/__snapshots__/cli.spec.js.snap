// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prettier-formatter-gitlab cli should create a code quality report file 1`] = `
"[
  {
    \\"type\\": \\"issue\\",
    \\"check_name\\": \\"prettier\\",
    \\"description\\": \\"Replace ·arg·· with arg\\",
    \\"severity\\": \\"minor\\",
    \\"fingerprint\\": \\"50ddf265a9c04daac13cec682b10f30f\\",
    \\"location\\": {
      \\"path\\": \\"test/dirty-2.js\\",
      \\"lines\\": {
        \\"begin\\": 3
      }
    }
  },
  {
    \\"type\\": \\"issue\\",
    \\"check_name\\": \\"prettier\\",
    \\"description\\": \\"Insert ;\\",
    \\"severity\\": \\"minor\\",
    \\"fingerprint\\": \\"4dde85a78b6345101364c1c0e336a483\\",
    \\"location\\": {
      \\"path\\": \\"test/dirty-2.js\\",
      \\"lines\\": {
        \\"begin\\": 5
      }
    }
  },
  {
    \\"type\\": \\"issue\\",
    \\"check_name\\": \\"prettier\\",
    \\"description\\": \\"Replace ⏎⏎⏎⏎⏎⏎function·foo(·arg··)·{⏎⏎ with function·foo(arg)·{\\",
    \\"severity\\": \\"minor\\",
    \\"fingerprint\\": \\"43695942319bb6406b30cd66ae57aede\\",
    \\"location\\": {
      \\"path\\": \\"test/dirty.js\\",
      \\"lines\\": {
        \\"begin\\": 3
      }
    }
  },
  {
    \\"type\\": \\"issue\\",
    \\"check_name\\": \\"prettier\\",
    \\"description\\": \\"Delete ⏎⏎⏎\\",
    \\"severity\\": \\"minor\\",
    \\"fingerprint\\": \\"6c97346ecaca37c5ccfd3663d7886069\\",
    \\"location\\": {
      \\"path\\": \\"test/dirty.js\\",
      \\"lines\\": {
        \\"begin\\": 12
      }
    }
  }
]"
`;
